### PR TITLE
Ignore -Werror

### LINKED ISSUE #735: Ignore/disable `-Werror`
Currently `-Werror` breaks type checking in the presence of warnings. That is, hover and other LSP features stop working or use stale data.

### PR DESCRIPTION
**Pull request by pepeiborra**
Fixes #735

### CODE REVIEW

File: src/Development/IDE/GHC/Compat.hs [hunk 1]
```diff
@@ -398,3 +399,13 @@ getConArgs = GHC.getConDetails
 
 getPackageName :: DynFlags -> Module.InstalledUnitId -> Maybe PackageName
 getPackageName dfs i = packageName <$> lookupPackage dfs (Module.DefiniteUnitId (Module.DefUnitId i))
+
+disableWarningsAsErrors :: DynFlags -> DynFlags
+disableWarningsAsErrors df =
+    flip gopt_unset Opt_WarnIsError $ foldl' wopt_unset_fatal df [toEnum 0 ..]
```
Comment (ndmitchell) [inline src/Development/IDE/GHC/Compat.hs:405]: Does `minBound` work here? Or even better, `enumerate` from Data.List.Extra?

File: test/exe/Main.hs [hunk 2]
```diff
@@ -3022,7 +3050,7 @@ mkRange :: Int -> Int -> Int -> Int -> Range
 mkRange a b c d = Range (Position a b) (Position c d)
 
 run :: Session a -> IO a
-run s = withTempDir $ \dir -> runInDir dir s
+run s = run' (const s)
```
Comment (ndmitchell) [inline test/exe/Main.hs:3053]: This change looks unrelated. How come? Whats the justification?

**Review by ndmitchell — APPROVED** [review]

File: test/exe/Main.hs (hunk 2 above)
Comment (pepeiborra) [reply test/exe/Main.hs:3053]: I spotted some duplicate code and took the chance to refactor it. Hlint? Human lint!

File: src/Development/IDE/GHC/Compat.hs (hunk 1 above)
Comment (pepeiborra) [reply src/Development/IDE/GHC/Compat.hs:405]: I copied this code from the source (GHC) as the function isn't exported in older GHC versions
